packages/nimble-components/src/anchor-button/index.ts

```diff
@@ -0,0 +1,111 @@
+import { attr } from '@microsoft/fast-element';
+import {
+    DesignSystem,
+    Anchor as FoundationAnchor,
+    anchorTemplate as template,
+    applyMixins,
+    DelegatesARIALink,
+    AnchorOptions
+} from '@microsoft/fast-foundation';
+import {
+    ButtonAppearance,
+    ButtonAppearanceVariant,
+    ButtonPattern,
+    ButtonWithAppearanceVariantPattern
+} from '../patterns/button/types';
+import { styles } from './styles';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-anchor-button': AnchorButton;
+    }
+}
+
+/**
+ * A nimble-styled anchor button
+ */
+export class AnchorButton
+    extends FoundationAnchor
+    implements ButtonPattern, ButtonWithAppearanceVariantPattern {
+    /**
+     * @public
+     * @remarks
+     * HTML Attribute: appearance
+     */
+    @attr
+    public appearance: ButtonAppearance = ButtonAppearance.outline;
+
+    /**
+     * @public
+     * @remarks
+     * HTML Attribute: appearance-variant
+     */
+    @attr({ attribute: 'appearance-variant' })
+    public appearanceVariant: ButtonAppearanceVariant;
+
+    /**
+     * @public
+     * @remarks
+     * HTML Attribute: content-hidden
+     */
+    @attr({ attribute: 'content-hidden', mode: 'boolean' })
+    public contentHidden = false;
+
+    /**
+     * @public
+     * @remarks
+     * HTML Attribute: disabled
+     */
+    @attr({ mode: 'boolean' })
+    public disabled = false;
+
+    private cachedTabIndex: number | undefined;
+
+    /**
+     * The <a> element (".control") in the shadow root is not automatically
+     * disabled when the host is. It does not support the disabled attribute.
+     * When the host is disabled, we must manually remove it from the tab order.
+     * We must also sync its disabled attribute with the host, because the
+     * shared button CSS expects the element with class "control" to get the
+     * disabled attribute when the host has it.
+     */
+    public disabledChanged(_prev: boolean | undefined, next: boolean): void {
```

Comment:
The native browser behavior is that links go into the disabled state when the href attribute is removed. It handles focus states and aria role changes, etc. Could we try that approach instead?

I think us keeping a disabled attribute could be okay but we should also try and make sure that you can disable the control with either the disabled attribute and / or removing href. That way applications that swap an `<a>` tag for a `<nimble-anchor>` tag can see expect the same behavior.

Reply:
I wasn't aware that clearing the `href` was the normal/standard way of disabling a link. Knowing that, I'm not sure we need a separate `disabled` attribute. Do you think it's worth the trouble? I think I prefer having one way of doing things. Also, I assume whatever we decide, it should be the same for both the anchor and the anchor button, i.e. we should not support `disabled` for just the anchor button.

Reply:
Sounds reasonable to me. We may want a story or a note on the href property in the story. to guide folks that way.

Reply:
After looking at how our design system treats `disabled` for buttons, I decided that the anchor button should support disabling via both `disabled` or missing `href`. Our button interface and shared styles include `disabled`, so from our design system's perspective, a button must support `disabled`. Doing so also makes it easier to swap between a regular button and an anchor button. The regular anchor still only supports disabling via removing `href`, just like `<a>`.